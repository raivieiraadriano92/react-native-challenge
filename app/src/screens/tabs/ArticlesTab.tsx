import { View } from "react-native";
import { TabScreen } from "src/navigation/types";

export const ArticlesTab: TabScreen<"Articles"> = () => {
  return <View />;
};
