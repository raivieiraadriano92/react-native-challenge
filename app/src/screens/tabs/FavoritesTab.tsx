import { FlatList } from "react-native";
import { StoryRow } from "src/components/StoryRow";
import { TabScreen } from "src/navigation/types";
import { useStoriesStore } from "src/store/storiesStore";

export const FavoritesTab: TabScreen<"Favorites"> = ({ navigation }) => {
  const storiesStore = useStoriesStore();

  return (
    <FlatList
      data={storiesStore.favorites}
      keyExtractor={(item) => item.objectID}
      renderItem={({ item }) => (
        <StoryRow
          isFavorite
          onPress={() =>
            navigation.navigate("Article", { story_url: item.story_url })
          }
          onPressFavorite={() => storiesStore.toggleFavorite(item)}
          story={item}
          testID={`story-${item.objectID}`}
        />
      )}
      showsVerticalScrollIndicator={false}
    />
  );
};
