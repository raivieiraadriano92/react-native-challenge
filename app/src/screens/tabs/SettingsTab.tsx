import { useTheme } from "@react-navigation/native";
import { Switch, Text, View } from "react-native";
import { TabScreen } from "src/navigation/types";
import { useStoriesStore } from "src/store/storiesStore";

export const SettingsTab: TabScreen<"Settings"> = () => {
  const storiesStore = useStoriesStore();

  const theme = useTheme();

  return (
    <View className="flex-1 p-6">
      <View className="space-y-3">
        <View className="space-y-1">
          <Text className="font-bold text-xl">Notifications</Text>
          <Text className="text-sm text-neutral-400">
            Stay updated with the latest tips, tricks, and news for your Android
            or iOS device! Enable notifications to never miss an article that
            helps you get the most out of your smartphone.
          </Text>
        </View>
        <View className="flex-row items-center justify-between">
          <Text className="text-sm">Enable notifications for Android</Text>
          <Switch
            testID="toggleAndroid"
            trackColor={{ false: "#767577", true: theme.colors.primary }}
            thumbColor={storiesStore.preferences.android ? "white" : "#f4f3f4"}
            ios_backgroundColor="#3e3e3e"
            onValueChange={storiesStore.toggleAndroid}
            value={storiesStore.preferences.android}
          />
        </View>
        <View className="flex-row items-center justify-between">
          <Text className="text-sm">Enable notifications for iOS</Text>
          <Switch
            testID="toggleIOS"
            trackColor={{ false: "#767577", true: theme.colors.primary }}
            thumbColor={storiesStore.preferences.ios ? "white" : "#f4f3f4"}
            ios_backgroundColor="#3e3e3e"
            onValueChange={storiesStore.toggleIOS}
            value={storiesStore.preferences.ios}
          />
        </View>
      </View>
    </View>
  );
};
