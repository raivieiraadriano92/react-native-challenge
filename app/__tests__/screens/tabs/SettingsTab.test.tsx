import { render } from "@testing-library/react-native";
import { SettingsTab } from "src/screens/tabs/SettingsTab";

const mockNavigationProps = {
  setOptions: jest.fn()
};

describe("<SettingsTab />", () => {
  it("Snapshot", () => {
    const tree = render(
      <SettingsTab navigation={mockNavigationProps} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
