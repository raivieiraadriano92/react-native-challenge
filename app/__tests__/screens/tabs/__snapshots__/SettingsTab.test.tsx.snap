// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsTab /> Snapshot 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "paddingBottom": 24,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 24,
      },
    ]
  }
>
  <View>
    <View>
      <Text
        style={
          [
            {
              "fontWeight": "700",
            },
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
          ]
        }
      >
        Notifications
      </Text>
      <Text
        style={
          [
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
            {
              "color": "#a3a3a3",
            },
            [
              {
                "marginTop": 4,
              },
            ],
          ]
        }
      >
        Stay updated with the latest tips, tricks, and news for your Android or iOS device! Enable notifications to never miss an article that helps you get the most out of your smartphone.
      </Text>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "justifyContent": "space-between",
          },
          [
            {
              "marginTop": 12,
            },
          ],
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        Enable notifications for Android
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="rgb(0, 122, 255)"
        style={
          [
            {
              "height": 31,
              "width": 51,
            },
            {
              "backgroundColor": "#3e3e3e",
              "borderRadius": 16,
            },
          ]
        }
        testID="toggleAndroid"
        thumbTintColor="#f4f3f4"
        tintColor="#767577"
        value={false}
      />
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "justifyContent": "space-between",
          },
          [
            {
              "marginTop": 12,
            },
          ],
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        Enable notifications for iOS
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="rgb(0, 122, 255)"
        style={
          [
            {
              "height": 31,
              "width": 51,
            },
            {
              "backgroundColor": "#3e3e3e",
              "borderRadius": 16,
            },
          ]
        }
        testID="toggleIOS"
        thumbTintColor="white"
        tintColor="#767577"
        value={true}
      />
    </View>
  </View>
</View>
`;
